Intermittent test failure: test_post_sets_rollout_dirty_on_population_change

Seen here: https://app.circleci.com/pipelines/github/mozilla/experimenter/57747/workflows/b02ffb5b-1f17-47dd-bfbd-853dedbb4635/jobs/335416


```
=================================== FAILURES ===================================
___ TestAudienceUpdateView.test_post_sets_rollout_dirty_on_population_change ___
[gw3] linux -- Python 3.13.5 /usr/local/bin/python3.13

self = <experimenter.nimbus_ui_new.tests.test_views.TestAudienceUpdateView testMethod=test_post_sets_rollout_dirty_on_population_change>

    def test_post_sets_rollout_dirty_on_population_change(self):
        experiment = NimbusExperimentFactory(
            is_rollout=True,
            status=NimbusExperiment.Status.LIVE,
            status_next=None,
            is_paused=False,
            publish_status=NimbusExperiment.PublishStatus.IDLE,
            population_percent=5,
            application=NimbusExperiment.Application.DESKTOP,
            channel=NimbusExperiment.Channel.BETA,
        )
    
        response = self.client.post(
            reverse("nimbus-new-update-audience", kwargs={"slug": experiment.slug}),
            {
                "channel": NimbusExperiment.Channel.BETA,
                "countries": [],
                "excluded_experiments_branches": [],
                "firefox_max_version": NimbusExperiment.Version.FIREFOX_84,
                "firefox_min_version": NimbusExperiment.Version.FIREFOX_83,
                "is_sticky": False,
                "languages": [],
                "locales": [],
                "population_percent": 10,
                "proposed_duration": 1,
                "proposed_enrollment": 1,
                "required_experiments_branches": [],
                "targeting_config_slug": NimbusExperiment.TargetingConfig.NO_TARGETING,
                "total_enrolled_clients": 0,
            },
        )
    
        self.assertEqual(response.status_code, 200)
        experiment.refresh_from_db()
>       self.assertTrue(experiment.is_rollout_dirty)
E       AssertionError: False is not true

experimenter/nimbus_ui_new/tests/test_views.py:2586: AssertionError
```

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-5630)

--- a/experimenter/experimenter/nimbus_ui_new/tests/test_forms.py
+++ b/experimenter/experimenter/nimbus_ui_new/tests/test_forms.py
@@ -1464,6 +1464,7 @@ class TestAudienceForm(RequestFormTestCase):
             population_percent=5,
             application=NimbusExperiment.Application.DESKTOP,
             channel=NimbusExperiment.Channel.BETA,
+            targeting_config_slug=NimbusExperiment.TargetingConfig.NO_TARGETING,
         )
 
         form = AudienceForm(

--- a/experimenter/experimenter/nimbus_ui_new/tests/test_views.py
+++ b/experimenter/experimenter/nimbus_ui_new/tests/test_views.py
@@ -2559,6 +2559,7 @@ class TestAudienceUpdateView(AuthTestCase):
             population_percent=5,
             application=NimbusExperiment.Application.DESKTOP,
             channel=NimbusExperiment.Channel.BETA,
+            targeting_config_slug=NimbusExperiment.TargetingConfig.NO_TARGETING,
         )
 
         response = self.client.post(

